internal/operands/template-validator/reconcile.go

```diff
@@ -114,15 +109,33 @@ func reconcileService(request *common.Request) error {
 		})
 }
 
-func reconcileDeployment(request *common.Request) error {
+func reconcileDeployment(request *common.Request) (common.ResourceStatus, error) {
 	validatorSpec := &request.Instance.Spec.TemplateValidator
 	image := common.GetTemplateValidatorImage()
 	deployment := newDeployment(request.Namespace, validatorSpec.Replicas, image)
 	addPlacementFields(deployment, &validatorSpec.Placement)
-	return common.CreateOrUpdateResource(request,
+	return common.CreateOrUpdateResourceWithStatus(request,
 		deployment,
 		func(newRes, foundRes controllerutil.Object) {
 			foundRes.(*apps.Deployment).Spec = newRes.(*apps.Deployment).Spec
+		},
+		func(res controllerutil.Object) common.ResourceStatus {
+			dep := res.(*apps.Deployment)
```

Comment:
Why not just read the `Status` of the deployment itself?
I think deployments provide `Available`, `Progressing` and `Degraded` conditions

Reply:
The `progressing` condition for Deployment is `true` even if all pods are deployed successfully. That is not the same as our `progressing` condition, which should be `false` when everything is deployed.

For now I would keep this code.